[bug] Windows + clang, Autotools, Libiconv failing compilation

### Describe the bug

I am working on windows, conan 2.26.2, and I would like to build an application using Qt.
I would like to use LLVM compiler clang for C++, with the MSVC runtime (as per the documentation example)

Following the doc, my profile looks like this
```
[settings]
arch=x86_64
build_type=Release
compiler=clang
compiler.version=21
compiler.cppstd=20
compiler.libcxx=libc++
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.runtime_version=v145
os=Windows

[conf]
tools.cmake.cmaketoolchain:generator=Ninja
tools.build:compiler_executables={"c": "clang", "cpp": "clang++"}
tools.compilation:verbosity=verbose
tools.microsoft.bash:subsystem=msys2
tools.microsoft.bash:path=C:\msys64\usr\bin\bash.exe


[buildenv]
PATH=+(path)C:/DevTools/LLVM/bin

[tool_requires]
ninja/[*]
```

And my conanfile.py is 
```py
class WriterRecipe(ConanFile):
    name = "writer"
    version = "0.1.0"
    settings = "os", "compiler", "build_type", "arch"

    def requirements(self):
        self.requires("qt/6.10.1")
        self.requires("spdlog/1.17.0")
        self.requires("sqlite3/3.51.0")
        self.requires("freetype/2.13.2")
        self.requires("glm/1.0.1")
        self.requires("doctest/2.4.12")

    def layout(self):
        cmake_layout(self)

    def generate(self):
        deps = CMakeDeps(self)
        deps.generate()
        tc = CMakeToolchain(self)
        tc.generator = "Ninja"
        tc.generate()
```

I am running the command `conan install . -pr clang --build=missing` and I cannot make it compile correctly, the issue stopping me at the moment being `libiconv/1.17`

```
libiconv/1.17: RUN: "/c/users/maxime.cabanal-duvil/.conan2/p/b/libic31ab7b587cb6a/b/src/configure" --disable-shared --enable-static --prefix=/ --bindir=${prefix}/bin --sbindir=${prefix}/bin --libdir=${prefix}/lib --includedir=${prefix}/include --oldincludedir=${prefix}/include 
conanvcvars.bat: Activating environment Visual Studio 18 - amd64 - winsdk_version=None - vcvars_ver=14.5
[vcvarsall.bat] Environment initialized for: 'x64'
configure: loading site script /etc/config.site
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... no
checking whether make supports nested variables... no
checking whether make sets $(MAKE)... (cached) no
checking for gcc... /c/users/maxime.cabanal-duvil/.conan2/p/b/libic31ab7b587cb6a/b/src/build-aux/compile clang-cl -nologo
checking whether the C compiler works... yes
checking for C compiler default output file name... conftest.exe
checking for suffix of executables... .exe
checking whether we are cross compiling... configure: error: in `/c/Users/Maxime.Cabanal-Duvil/.conan2/p/b/libic31ab7b587cb6a/b/build-release':
configure: error: cannot run C compiled programs.
If you meant to cross compile, use `--host'.
See `config.log' for more details
```

I am not trying to cross-compile anything, and I do not understand why does it choose to use `clang-cl` instead of `clang` when I explicitly told it to use the latter.

The profile being from the documentation, I do not know what I am missing here.



### How to reproduce it

_No response_

--- a/recipes/libiconv/all/conanfile.py
+++ b/recipes/libiconv/all/conanfile.py
@@ -46,7 +46,9 @@ class LibiconvConan(ConanFile):
 
     @property
     def _msvc_tools(self):
-        return ("clang-cl", "llvm-lib", "lld-link") if self._is_clang_cl else ("cl", "lib", "link")
+        compilers = self.conf.get("tools.build:compiler_executables", default={})
+        compiler = compilers.get("c") or compilers.get("cpp")
+        return (compiler or "clang-cl", "llvm-lib", "lld-link") if self._is_clang_cl else ("cl", "lib", "link")
 
     def export_sources(self):
         export_conandata_patches(self)
@@ -94,11 +96,13 @@ class LibiconvConan(ConanFile):
         env = tc.environment()
         if is_msvc(self) or self._is_clang_cl:
             cc, lib, link = self._msvc_tools
+            if cc.endswith("clang-cl"):
+                cc = f"{cc} -nologo"
             build_aux_path = os.path.join(self.source_folder, "build-aux")
             lt_compile = unix_path(self, os.path.join(build_aux_path, "compile"))
             lt_ar = unix_path(self, os.path.join(build_aux_path, "ar-lib"))
-            env.define("CC", f"{lt_compile} {cc} -nologo")
-            env.define("CXX", f"{lt_compile} {cc} -nologo")
+            env.define("CC", f"{lt_compile} {cc}")
+            env.define("CXX", f"{lt_compile} {cc}")
             env.define("LD", link)
             env.define("STRIP", ":")
             env.define("AR", f"{lt_ar} {lib}")

--- a/recipes/libiconv/all/conanfile.py
+++ b/recipes/libiconv/all/conanfile.py
@@ -96,7 +96,7 @@ class LibiconvConan(ConanFile):
         env = tc.environment()
         if is_msvc(self) or self._is_clang_cl:
             cc, lib, link = self._msvc_tools
-            if cc.endswith("clang-cl"):
+            if cc.endswith("cl"):
                 cc = f"{cc} -nologo"
             build_aux_path = os.path.join(self.source_folder, "build-aux")
             lt_compile = unix_path(self, os.path.join(build_aux_path, "compile"))

